lib/v2/bloomberg/utils.js

```diff
@@ -0,0 +1,289 @@
+const cheerio = require('cheerio');
+const { parseString } = require('xml2js');
+const path = require('path');
+
+const { parseDate } = require('@/utils/parse-date');
+const got = require('@/utils/got');
+const { art } = require('@/utils/render');
+
+const rootUrl = 'https://www.bloomberg.com/feeds';
+const sel = 'script[data-component-props="ArticleBody"], script[data-component-props="FeatureBody"]';
+const apiEndpoints = {
+    articles: {
+        url: 'https://www.bloomberg.com/javelin/api/foundation_transporter/',
+        sel,
+    },
+    features: {
+        url: 'https://www.bloomberg.com/javelin/api/foundation_feature_transporter/',
+        sel,
+    },
+    audio: {
+        url: 'https://www.bloomberg.com/news/audio/',
+        sel: 'script#__NEXT_DATA__',
+    },
+    videos: {
+        url: 'https://www.bloomberg.com/news/videos/',
+        sel: '.footer-container + script',
+    },
+    newsletters: {
+        url: 'https://www.bloomberg.com/news/newsletters/',
+        sel,
+    },
+    'photo-essays': {
+        url: 'https://www.bloomberg.com/javelin/api/photo-essay_transporter/',
+        sel: 'script[type = "application/json"][data-component-props]',
+    },
+};
+const headers = {
+    accept: 'application/json',
+    'cache-control': 'no-cache',
+    referer: 'https://www.bloomberg.com',
+};
+
+const typeRegex = /\/([\w-]*?)\/(\d{4}-\d{2}-\d{2}\/.*)/;
+const capRegex = /<p>|<\/p>/g;
+const emptyRegex = /<p\b[^>]*>(&nbsp;|\s)<\/p>/g;
+
+const parseNewsList = async (url, ctx) => {
+    const resp = await got(url);
+    const {
+        urlset: { url: urls },
+    } = await new Promise((resolve, reject) => {
+        parseString(resp.data, (err, result) => {
+            if (err) {
+                reject(err);
+                return;
+            }
+            resolve(result);
+        });
+    });
+
+    return urls.slice(0, ctx.query.limit ? parseInt(ctx.query.limit) : 50).map((u) => {
+        const news = u['news:news'][0];
+        const item = {
+            link: u.loc[0],
+            title: news['news:title'][0],
+            pubDate: parseDate(news['news:publication_date'][0]),
+        };
+        return item;
+    });
```

Comment:
You can use `cheerio` to handle xml as well
```suggestion
    const $ = cheerio.load(resp.data, {
        xml: {
            xmlMode: true,
        },
    });
    const urls = $('urlset url');
    return urls.toArray().slice(0, ctx.query.limit ? parseInt(ctx.query.limit) : 50).map((u) => {
        u = $(u);
        const item = {
            title: u.find('news\\:title').text(),
            link: u.find('loc').text(),
            pubDate: parseDate(u.find('news\\:publication_date').text()),
        };
        return item;
    });
```

Reply:
looking good for native async support